purchase_lead_hour/models/stock_orderpoint.py

```diff
@@ -0,0 +1,28 @@
+# Copyright 2022 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+from dateutil import relativedelta
+
+from odoo import models
+
+
+class StockOrderpoint(models.Model):
+    _inherit = "stock.warehouse.orderpoint"
+
+    def _get_product_context(self):
+        # Since product.supplierinfo delay is now a float, we need to get the
+        # available qty as of now + delay
```

Comment:
But here you only use your new hook `_get_next_availability_date` to compute the forecasted qty, and the forecasted date will remain unchanged if we override it. That's a blocking issue.

`_get_product_context` was using `lead_days_date` (computed from today in `_compute_lead_days`).
As you are overriding this lead date only in `_get_product_context`, maybe you will get the right qty, but the `lead_days_date` is still returning the std date.

We need a hook in Odoo core to replace the call to `fields.Date.today()` in `_compute_lead_days` so you can put the date/datetime you want, so you'll get coherent regarding the lead date and the qty. Then your `_get_product_context` could be removed.